Keen/Query/Queries.cs

```diff
@@ -2,66 +2,93 @@
 using Newtonsoft.Json.Linq;
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
-using System.Dynamic;
-using System.IO;
 using System.Linq;
-using System.Net.Http;
-using System.Text;
 using System.Threading.Tasks;
 
+
 namespace Keen.Core.Query
 {
     /// <summary>
     /// Queries implements the IQueries interface which represents the Keen.IO Query API methods.
     /// </summary>
     internal class Queries : IQueries
     {
-        private IProjectSettings _prjSettings;
-        private string _serverUrl;
+        private readonly IKeenHttpClient _keenHttpClient;
+        private readonly string _queryRelativeUrl;
+        private readonly string _key;
```

Comment:
In general defensive copies are nice, but since strings are immutable it's a little bit less of a concern. Once I started thinking about how we'd use the KeenClient to create Access Keys, I sort of started leaning back toward just requesting the key at message dispatch time.